### PR TITLE
Update co group auth exp

### PR DESCRIPTION
**Pull request by tpharrison**
## Summary
The `codeowners` file in vets-api is referencing the `vfs-authenticated-experience` team. This team contains more than just BE engineers so non-engineers keep getting email notifications.  I created a `vfs-authenticated-experience-backend` team for just the BE engineers and updating the `codeowners` file to use this team instead.

## Related issue(s)
https://github.com/department-of-veterans-affairs/va.gov-team/issues/66007

## Testing done
N/A

### CODE REVIEW

**Comment by tpharrison** [comment]
@department-of-veterans-affairs/backend-review-group Getting this error message...
> New entry on line 75 of CODEOWNERS does not include https://github.com/orgs/department-of-veterans-affairs/teams/backend-review-group.

But the entry is there.  Not sure how to proceed - any advice?  Thanks!

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -147,9 +147,9 @@ app/mailers/transactional_email_mailer.rb @department-of-veterans-affairs/my-edu
...
 app/mailers/views/ch31_submissions_report.html.erb @department-of-veterans-affairs/vsa-ebenefits @department-of-veterans-affairs/va-api-engineers @department-of-veterans-affairs/backend-review-group
 app/mailers/views/dependents_application_failure.erb @department-of-veterans-affairs/vsa-ebenefits @department-of-veterans-affairs/va-api-engineers @department-of-veterans-affairs/backend-review-group
-app/mailers/views/direct_deposit.html.erb @department-of-veterans-affairs/vfs-authenticated-experience
-app/mailers/views/failed_claims_report.erb @department-of-veterans-affairs/vfs-authenticated-experience
-app/mailers/views/hca_submission_failure.html.erb @department-of-veterans-affairs/vfs-authenticated-experience
+app/mailers/views/direct_deposit.html.erb @department-of-veterans-affairs/vfs-authenticated-experience-backend
+app/mailers/views/failed_claims_report.erb @department-of-veterans-affairs/vfs-authenticated-experience-backend
+app/mailers/views/hca_submission_failure.html.erb @department-of-veterans-affairs/vfs-authenticated-experience-backend
```
Comment (jerekshoe) [inline .github/CODEOWNERS:152]: Are you keeping this file?

Comment (tpharrison) [reply .github/CODEOWNERS:152]: No, but I was planning on reassigning code ownership in subsequent PRs.  Is this an issue?

Comment (jerekshoe) [reply .github/CODEOWNERS:152]: No, just wanted to verify that you weren’t planning on changing the file assignments here, just the team name

**Review by jerekshoe — DISMISSED** [review]

**Pull request merged**